### PR TITLE
node: Adding WithHost options to settings section

### LINKED ISSUE #299: node: add WithHost Option
WithHost option is needed to adjust Host param to the node. 
It is required in order for the [Test Swamp](https://github.com/celestiaorg/celestia-node/issues/262) to link+connect nodes for further testing

### PR DESCRIPTION
**Pull request by Bidon15**
Depends on #292 

Removing append from overrides func
Edit node.go
Adding 1 test for checking withHost functionality

Closes #299

### CODE REVIEW

File: node/node.go [hunk 1]
```diff
@@ -82,9 +82,9 @@ func New(tp Type, store Store, options ...Option) (*Node, error) {
 
 	switch tp {
 	case Bridge:
-		return newNode(bridgeComponents(cfg, store), fxutil.Options(s.overrides()...))
+		return newNode(bridgeComponents(cfg, store), fxutil.Options(s.overrides()))
```
Comment (Wondertan) [inline node/node.go:85]: You can remove `fxutil.Options` here

File: node/node.go [hunk 2]
```diff
@@ -82,9 +82,9 @@ func New(tp Type, store Store, options ...Option) (*Node, error) {
...
 	case Light:
-		return newNode(lightComponents(cfg, store), fxutil.Options(s.overrides()...))
+		return newNode(lightComponents(cfg, store), fxutil.Options(s.overrides()))
```
Comment (Wondertan) [inline node/node.go:87]: and here

**Review by Wondertan — APPROVED** [review]

File: CHANGELOG-PENDING.md [hunk 3]
```diff
@@ -21,6 +21,7 @@ Month, DD, YYYY
 - [chore: bump deps #297](https://github.com/celestiaorg/celestia-node/pull/297) [@Wondertan](https://github.com/Wondertan)
 - [workflows/lint: update golangci-lint to v1.43 #308](https://github.com/celestiaorg/celestia-node/pull/308) [@Wondertan](https://github.com/Wondertan)
 - [feat(node): extract overrides from Config into Settings #292](https://github.com/celestiaorg/celestia-node/pull/292) [@Wondertan](https://github.com/Wondertan)
+- [node: Adding WithHost options to settings section #301](https://github.com/celestiaorg/celestia-node/pull/301) @Bidon15
```
Comment (renaynay) [inline CHANGELOG-PENDING.md:24]: Link to your github

**Review by renaynay — APPROVED** [review]

**Review by renaynay — COMMENTED** [review]
Actually - how are these options intended to be used? Where is the CLI flag intended to trigger this option override? Same for #302 .

**Comment by Wondertan** [comment]
They are not going to be triggered by CLI, but by TestSwamp. TestSwamp needs to overwrite those with custom testing values.

**Comment by Bidon15** [comment]
> Actually - how are these options intended to be used? Where is the CLI flag intended to trigger this option override? Same for #302 .

We need these flags to initialise the node with a predefined network and coreclient created by test swamp

File: CHANGELOG-PENDING.md (hunk 3 above)
Comment (Bidon15) [reply CHANGELOG-PENDING.md:24]: I'll fix it in #302

**Pull request merged**